instant-xml-macros/src/ser.rs

```diff
@@ -56,6 +61,8 @@ fn serialize_enum(
             ) -> Result<(), instant_xml::Error> {
 		serializer.write_str(match self { #variants })
             }
+
+            const KIND: ::instant_xml::Kind = ::instant_xml::Kind::Scalar;
```

Comment:
This is arguably a separate change, which also merits its own commit.

Reply:
This is part of fixing the serialization logic, though. This constant is part of the trait.

Reply:
I guess it would be okay to have the required fixes in the same commit as the additional test(s).